src/main/java/org/mskcc/kickoff/lims/SampleInfoExome.java

```diff
@@ -208,6 +226,26 @@ private void processCaptureInfo(DataRecordManager drm, DataRecord rec, User apiU
         this.BAIT_VERSION = this.CAPTURE_BAIT_SET;
     }
 
+    private List<List<Map<String, Object>>> getKapaAgilentCaptureProtocolFromPrevSamps(DataRecordManager drm, DataRecord sampleRec, User apiUser, String kapaProtocolVersion) {
+        Set<List<Map<String, Object>>> kapaFieldsSet = new HashSet<>();
```

Comment:
is it set on purpose? because then I see it's used as list but we will loose duplicates anyway. But I guess it's not relevant?

Reply:
yes, `Set` is used on purpose. for one thing, i think including duplicates makes no sense for downstream code where only the first item in the list is used. For another, I think there is a bug on veloxapi's getFieldsForDescendantsOfType method where it can return a list of size one even when no record is found. if set or such checking is not used, we may end up with list like [[], [], [record1], [record1]], when the desired result shall be [[record1]].